src/web/installer.go

```diff
@@ -139,8 +144,31 @@ func startup() {
 	}()
 }
 
-func awaitStartup() { // blocks and only returns once startup is done
-	<-ready
+// download the installer, if an error occurs keep trying again after duration (blocking)
+func downloadUntilSuccess(version InstallerVersion, duration time.Duration) []byte {
+	var attempts int
+	exe, err := version.fetchFile()
+	for err != nil {
+		attempts++
+		fmt.Fprintf(os.Stderr, "Error downloading %s Installer after %d attempts: %s\n", version.getEXT(), attempts, err.Error())
+		time.Sleep(duration)
+		exe, err = version.fetchFile()
+	}
+	return exe
+}
+
+// blocks and only returns once startup is done or timeout is complete
+// true if startup is done
+func awaitStartup(timeout time.Duration) bool {
+	ticker := time.NewTicker(timeout)
+	for {
```

Comment:
This doesn't need to be in a for loop, you can just have the select as-is without it being wrapped in a for